astroscrappy/astroscrappy.pyx

```diff
@@ -339,21 +401,34 @@ def detect_cosmics(indat, inmask=None, float sigclip=4.5, float sigfrac=0.3,
         if cleantype == 'median':
             # Unmasked median filter
             crinds = crmask > 0
-            cleanarr[crinds] = m5[crinds]
-            del m5
+            cleanarr[crinds] = m5_nobkg[crinds]
+            del m5_nobkg
+            if var is not None:
```

Comment:
Similarly here, I think you can have a noise and just median of the cleanarr which would simplify this whole block considerably.